brave/brave-core
# Implement and expose diagnostic logging for Brave Rewards in the UI

Resolves https://github.com/brave/brave-browser/issues/9533

## Submitter Checklist:

- Verified that these changes build without errors on
- Verified that these changes pass automated tests (unit, browser, security tests) on

## Test Plan:

- Confirm `Rewards.log` is written to the user profile `Default` folder
- Confirm `Rewards.log` is tailed to the last 20,000 lines when restarting the browser
- Confirm `Rewards.log` is tailed to the last 20,000 lines when the log exceeds 10mb
- Confirm `Rewards.log` does not degrade performance when tailed
- Confirm user can auto-refresh log in brave://rewards-internals
- Confirm user can refresh log in brave://rewards-internals
- Confirm user can refresh log in brave://rewards-internals
- Confirm user can clear log in brave://rewards-internals
- Confirm cleared logs are not shown when refreshing the log
- Confirm user can download log

## Reviewer Checklist:


## After-merge Checklist:

  changes has landed on.
